gtsam/inference/doc/ClusterTree.ipynb

```diff
@@ -0,0 +1,50 @@
+{
+  "cells": [
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "clustertree_intro_md"
+      },
+      "source": [
+        "# ClusterTree"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "clustertree_desc_md"
+      },
+      "source": [
+        "A `ClusterTree` is a more general structure than an `EliminationTree` or `JunctionTree`. It represents a tree where each node (a 'Cluster') contains a subset of factors from an original factor graph. The key property is that the tree must be 'family preserving', meaning each original factor must belong entirely within a single cluster.\n",
```

Comment:
Could we replace all references  to classes in the docs with links to their notebook? Wondering whether this could be done with a custom “role” in Myst: I know you played around with that?